### PR TITLE
Pub: Fix package project duplicates and other issues

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Please see the commit messages for details.

### CODE REVIEW

File: analyzer/src/funTest/kotlin/managers/PubFunTest.kt [hunk 1]
```diff
@@ -63,8 +63,6 @@ class PubFunTest : WordSpec() {
                     val vcsPath = vcsDir.getPathToRoot(workingDir)
                     val expectedResult = patchExpectedResult(
                         expectedResultFile,
-                        custom = mapOf("pub-project" to "pub-${workingDir.name}"),
```
Comment (sschuberth) [inline analyzer/src/funTest/kotlin/managers/PubFunTest.kt:66]: Could you briefly explain in the commit message *why* they are unnecessary? Like, does e.g. the "pub-project" string simply not occur in the files?

File: analyzer/src/funTest/assets/projects/external/dart-http-expected-output.yml [hunk 2]
```diff
@@ -1286,7 +1286,7 @@ packages:
   - "Dart Team"
```
Comment (oheger-bosch) [inline analyzer/src/funTest/assets/projects/external/dart-http-expected-output.yml:1286]: Commit message: The body is not a full sentence; I think the first word is wrong.

**Review by oheger-bosch — COMMENTED** [review]
Changes look good to me.

**Review by sschuberth — DISMISSED** [review]

**Review by oheger-bosch — APPROVED** [review]

**Comment by mnonnenmacher** [comment]
I have added a fix for a subversion test.

**Review by oheger-bosch — APPROVED** [review]

**Comment by mnonnenmacher** [comment]
Merged, as test failures in `DotNetFunTest` are unrelated to the change.

**Pull request merged**